### PR TITLE
feat(quantic): searchbox without tributejs dependency

### PR DESCRIPTION
**Pull request by nathanlb**
The searchbox was behaving badly as evidence by this video where i enter text then press enter which replaces my query with the first suggestion.

https://user-images.githubusercontent.com/16785453/112149873-d4c5b400-8bb5-11eb-9a8e-9a43079087c6.mov

While attempting to fix it, which proved difficult, I decided it would be best to remove the tributejs dependency altogether and redo the searchbox from scratch, giving us much more control over its behaviour.

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/searchBox/searchBox.js [hunk 1]
```diff
@@ -1,43 +1,75 @@
...
   /** @type {import("coveo").Unsubscribe} */
   unsubscribe;
-  /** @type {any} */
-  tribute;
-  tributeLoaded = false;
+  /** @type {number} */
+  selectionIndex = -1;
+  /** @type {HTMLInputElement} */
+  input;
+  /** @type {import("lwc").HTMLElementTheGoodPart} */
```
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/searchBox/searchBox.js:38]: HTMLElementTheGoodPart ? What?

File: packages/quantic/force-app/main/default/lwc/resultLink/resultLink.html [hunk 2]
```diff
@@ -1,3 +1,3 @@
 <template>
-    <a href={result.clickUri}>{result.title}</a>
+    <a target="_blank" href={result.clickUri}>{result.title}</a>
```
[the remark this answers is not in the document]
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/resultLink/resultLink.html:2]: Fair enough. I was influenced by my preference for opening in another tab. It was out of scope anyway. I'll revert it.

File: packages/quantic/force-app/main/default/lwc/searchBox/searchBox.js (hunk 1 above)
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/searchBox/searchBox.js:38]: Yeah it's a lwc type that bring in stuff from HTMLElement and some additions from lwc to reference attributes of "native" web elements.
References to this type existed before this PR as well :)

**Review by lbergeron — APPROVED** [review]
Looks good to me

**Pull request merged**